documentation/getting-started.adoc

```diff
@@ -0,0 +1,245 @@
+
+
+=== Cluster Controller
+
+Strimzi uses a component called the Cluster Controller to deploy and manage Kafka (including Zookeeper) and Kafka Connect
+clusters. The Cluster Controller is deployed as a process running inside your Kubernetes or OpenShift cluster. To deploy a
+Kafka cluster, a ConfigMap with the cluster configuration has to be created. Based on the information in that ConfigMap,
+the Cluster Controller will deploy a corresponding Kafka cluster. By default, the ConfigMap needs to be labeled with
+following labels:
+
+[source,yaml]
+strimzi.io/type: kafka
+strimzi.io/kind: cluster
+
+and contain the cluster configuration in a specific format.
+
+The Strimzi project contains example YAML files which make deploying of Cluster Controller easier.
+
+==== Deploying to Kubernetes
+
+To deploy the Cluster Controller on Kubernetes, following command should be executed:
+
+[source]
+kubectl create -f resources/kubernetes/cluster-controller.yaml
+
+To verify whether the Cluster Controller has been deployed successfully, the Kubernetes Dashboard or the following
+command can be used:
+
+[soruce]
+kubectl describe all
+
+==== Deploying to OpenShift
+
+To deploy the Cluster Controller on OpenShift, following command should be executed:
+
+[source]
+oc create -f resources/openshift/cluster-controller-with-template.yaml
+
+To verify whether the Cluster Controller has been deployed successfully, the OpenShift console or the following command
+can be used:
+
+[soruce]
+oc describe all
+
+=== Kafka broker
+
+Strimzi uses StatefulSets feature to deploy Kafka brokers on Kubernetes/OpenShift.
+With StatefulSets, the pods receive a unique name and network identity and that makes it easier to identify the
+individual Kafka broker pods and set their identity (broker ID). The deployment uses **regular** and **headless**
+services:
+
+- regular services can be used as bootstrap servers for Kafka clients
+- headless services are needed to have DNS resolve the pods IP addresses directly
+
+Together with Kafka, Strimzi also installs a Zookeeper cluster and configures the Kafka brokers to connect to it. The
+Zookeeper cluster also uses StatefulSets.
+
+Strimzi provides two flavors of Kafka broker deployment: **ephemeral** and **persistent**.
+
+The **ephemeral** flavour is suitable only for development and testing purposes and not for production. The
+ephemeral flavour uses `emptyDir` volumes for storing broker information (Zookeeper) and topics/partitions
+(Kafka). Using `emptyDir` volume means that its content is strictly related to the pod life cycle (it is
+deleted when the pod goes down). This makes the in-memory deployment well-suited to development and testing because
+you don't have to provide persistent volumes.
+
+The **persistent** flavour uses PersistentVolumes to store Zookeeper and Kafka data. The PersistentVolume is
+acquired using a PersistentVolumeClaim – that makes it independent on the actual type of the PersistentVolume. For
```

Comment:
independent of.

Also, but not for this PR, I wonder whether we should link stuff like PersistentVolme, StatefulSet to the corresponding Kubernetes docs?

Reply:
I think that the links have been quite unstable lately as there is a lot of effort to improve the Kubernetes docu. So not sure it is worth it.

Reply:
OK, that's a good enough reason not to do it!